collector/library/pod.js

```diff
@@ -224,15 +210,57 @@ class Pod {
      * 
      */
     async submitDataToLogstash({ data, batch, }) {
-        const body = JSON.stringify({ message: data.toString() });
+        const dataAsString = data.toString();
+        let lines;
+
+        if (this.remainder.length > 0) {
+            lines = (this.remainder + dataAsString).split('\n');
+            this.remainder = '';
+        } else {
+            lines = dataAsString.split('\n');
+        }
 
-        await fetch('http://logstash:5000/', {
-            method: 'post',
-            body,
-            headers: { 'Content-Type': 'application/json' },
-        });
+        for (let i = 0; i < lines.length; i++) {
```

Comment:
simplify the flow 

```
for (let i=0; i < lines.length - 1; i++) { // we always skip the last line - it is for reminder
...
}
this.remainder = lines[lines.length - 1];

Reply:
Any concrete suggestions here? I cannot think of anything in particular